### PR TITLE
expression: rewrite builtin function: ELT

### PR DESCRIPTION
**Pull request by spongedu**
for #4080

### CODE REVIEW

**Comment by zz-jason** [comment]
modify `TestElt` in `expression/builtin_string_test.go` and check `f.isDeterministic()`

File: expression/builtin_string.go [hunk 1]
```diff
@@ -2050,39 +2050,48 @@ type eltFunctionClass struct {
...
+	for i := 1; i < len(args); i++ {
+		argTps = append(argTps, tpString)
+	}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, argTps...)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	for _, arg := range args[1:] {
+		argType := arg.GetType()
+		//setBinFlagOrBinStr(argType, bf.tp)
```
Comment (hanfei1991) [inline expression/builtin_string.go]: Why comment this ?

Comment (spongedu) [inline expression/builtin_string.go]: @hanfei1991 I am not sure whether 'Charset', 'Collate' and 'Flag' changes according to parameters in builtin 'ELT'. I am trying to feature out how it works in Mysql. Do you have any ideas about this ?

Comment (XuHuaiyu) [inline expression/builtin_string.go]: ELT returns a binary string, only when at least one argument is a binary string.

Comment (spongedu) [inline expression/builtin_string.go]: @XuHuaiyu  OK, thx.

File: expression/builtin_string.go [hunk 2]
```diff
@@ -2236,7 +2236,7 @@ func (c *eltFunctionClass) getFunction(args []Expression, ctx context.Context) (
 	}
 	for _, arg := range args[1:] {
 		argType := arg.GetType()
-		//setBinFlagOrBinStr(argType, bf.tp)
+		setBinFlagOrBinStr(argType, bf.tp)
```
Comment (XuHuaiyu) [inline expression/builtin_string.go]: this is not consistent with MySQL
``` sql
mysql> select elt(2, "abc", 1);
Field   1:  `elt(2, "abc", 1)`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       VAR_STRING
Collation:  utf8_general_ci (33)
Length:     9
Max_length: 1
Decimals:   31
Flags:


+------------------+
| elt(2, "abc", 1) |
+------------------+
| 1                |
+------------------+
1 row in set (0.01 sec)
```

**Comment by zz-jason** [comment]
@spongedu plz resolve conflicts

**Pull request merged**